runner.py

```diff
@@ -398,29 +398,35 @@ def setup_services(self):
 
             if 'environment' in service:
                 for docker_env_var in service['environment']:
-                    if not self._allow_unsafe and re.search(r'^[A-Z_]+$', str(docker_env_var)) is None:
+                    # In a compose file env vars can be defined with a "=" not as a dict.
+                    if '=' in docker_env_var:
+                        env_key, env_value = docker_env_var.split('=')
+                    else:
+                        env_key, env_value = str(docker_env_var), str(service['environment'][docker_env_var])
+
+                    if not self._allow_unsafe and re.search(r'^[A-Z_]+$', env_key) is None:
                         if self._skip_unsafe:
                             warn_message= arrows(f"Found environment var key with wrong format. \
-                                 Only ^[A-Z_]+$ allowed: {docker_env_var} - Skipping")
+                                 Only ^[A-Z_]+$ allowed: {env_key} - Skipping")
                             print(TerminalColors.WARNING, warn_message, TerminalColors.ENDC)
                             continue
                         raise RuntimeError(f"Docker container setup environment var key had wrong format. \
-                            Only ^[A-Z_]+$ allowed: {docker_env_var} - Maybe consider using --allow-unsafe \
+                            Only ^[A-Z_]+$ allowed: {env_key} - Maybe consider using --allow-unsafe \
                                 or --skip-unsafe")
 
                     if not self._allow_unsafe and \
-                        re.search(r'^[a-zA-Z_]+[a-zA-Z0-9_-]*$', str(service['environment'][docker_env_var])) is None:
+                        re.search(r'^[a-zA-Z0-9_]+[a-zA-Z0-9_-]*$', env_value) is None:
```

Comment:
We need to discuss this. Is this a security problem? @ArneTR

Reply:
The security issue is that you could reference a variable from the host. Checking both key and value with a whitelist is what I choose.
You specify this yourself in the comment in the changed files ...? Or am I missing something?

Reply:
Currently every value needs to start with a char or _. But let's say you have a password that starts with number or has special characters. For example you could have an env var:
```
MYSQL_PASSWD=123$%^FOO
```
which will currently fail because we only allow `r'^[a-zA-Z_]+[a-zA-Z0-9_-]*$'` I now changed this to also allow 0-9 as first char as my std password for dev  work is `123`. I am not sure how to handle the value part from the key. Because if we allow anything people could do `$(SOMETHING)` or similar but if we are too restrictive we might run into issues with certain parameters. For now I would also allow 0-9 for the first char and we can see if someone complains. I would rather have it a little more secure per default.